### ISSUE TITLE
Stream2 ?

### ISSUE THREAD

**Issue by triccardi-systran** [felixge/node-combined-stream#21]
Wouldn't it be safer and easier to use nodejs 0.10 Stream2 API?
See http://nodejs.org/api/stream.html#stream_api_for_stream_implementors

**Comment by joepie91** [comment]
I don't know if there are any particular plans to update `combined-stream` to support Streams2, but I've just published a drop-in replacement that supports Streams2: https://www.npmjs.com/package/combined-stream2

**Comment by alexindigo** [comment]
This is cool, we'll play with it. Thank  you.

**Comment by triccardi-systran** [comment]
And what about https://www.npmjs.com/package/multistream ?
Why rewrite another one? It seems already used, and has tests.

**Comment by joepie91** [comment]
@triccardi-systran `multistream` does not take Buffers, and has a different API. My goal was a (mostly) API-compatible implementation :)

**Comment by joepie91** [comment]
@apechimp Was any resolution reached on this?

**Comment by apechimp** [comment]
Nope. The streams 2 based libraries have been mentioned and this library will not switch to streams 2 in the foreseeable future.

**Comment by triccardi-systran** [comment]
This should be plainly stated

**Comment by triccardi-systran** [comment]
For example, highly used modules `form-data` and `request` depend on `combined-stream`, and thus are limited by Stream 1 API.

**Comment by apechimp** [comment]
That readme change is a welcome PR.

**Comment by alexindigo** [comment]
With all the streams2 libraries all around, they all miss focus is preserving main use case – support of `node-form-data` and `request`, looks like we need to give that a crack here.

**Comment by joepie91** [comment]
@alexindigo What functionality is required for `form-data`/`request` that got lost in `combined-stream2`?
